segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -568,31 +568,22 @@ public void initialize() throws Exception {
                         val snapshotFileName = NameUtils.getSystemJournalSnapshotFileName(containerId, snapshotInfo.getEpoch(), snapshotInfo.getSnapshotId());
                         log.debug("SystemJournal[{}] Snapshot info read. {} pointing to {}", containerId, snapshotInfo, snapshotFileName);
 
-                        // Step 2: Validate.
-                        return checkSnapshotExists(snapshotFileName)
-                                // Step 3: Read contents.
-                                .thenComposeAsync(v -> getContents(snapshotFileName), executor)
-                                // Step 4: Deserialize and return.
-                                .thenApplyAsync(contents -> readSnapshotRecord(snapshotInfo, contents), executor);
+                        // Step 2: Validate. and Read contents.
+                        return getContents(snapshotFileName, false)
+                                // Step 3: Deserialize and return.
+                                .thenApplyAsync(contents -> readSnapshotRecord(snapshotInfo, contents), executor)
+                                .exceptionally( e -> {
+                                    throw new CompletionException(new IllegalStateException(
+                                            String.format("Chunk pointed by SnapshotInfo must exist. Missing chunk %s", snapshotFileName),
```

Comment:
Please, can you add a test by inducing an exception being thrown from `readSnapshotRecord` to see if it is being handled properly?

Reply:
We are testing all sorts of failures pretty extensively in 
https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/SystemJournalOperationsTests.java

These tests already cover the scenario.